MapleServer2/PacketHandlers/Game/RockPaperScissorsHandler.cs

```diff
@@ -0,0 +1,320 @@
+                {
+                    result = RPSResult.Draw;
+                }
+                else
+                {
+                    result = RPSResult.Lose;
+                }
+            }
+            else
+            {
+                if (opponent.RPSSelection == RockPaperScissorsChoice.Rock)
+                {
+                    result = RPSResult.Lose;
+                }
+                else if (opponent.RPSSelection == RockPaperScissorsChoice.Paper)
+                {
+                    result = RPSResult.Win;
+                }
+                else
+                {
+                    result = RPSResult.Draw;
+                }
+            }
+
+            RPS rpsEvent = DatabaseManager.Events.FindRockPaperScissorsEvent();
+            if (rpsEvent is null)
+            {
+                return;
+            }
+
+            Item voucher = session.Player.Inventory.Items.Values.FirstOrDefault(x => x.Id == rpsEvent.VoucherId);
+            if (voucher is null)
+            {
+                return;
+            }
+
+            session.Player.Inventory.ConsumeItem(session, voucher.Uid, 1);
+
+            GameEventUserValue dailyMatches = GameEventHelper.GetUserValue(session.Player, rpsEvent.Id, TimeInfo.Tomorrow(), GameEventUserValueType.RPSDailyMatches);
+            if (!int.TryParse(dailyMatches.EventValue, out int dailyMatchCount))
+            {
+                dailyMatchCount = 0;
+            }
+
+            dailyMatchCount++;
+
+            dailyMatches.EventValue = dailyMatchCount.ToString();
+
+            DatabaseManager.GameEventUserValue.Update(dailyMatches);
+            session.Send(GameEventUserValuePacket.UpdateValue(dailyMatches));
+            session.Send(RockPaperScissorsPacket.MatchResults(result, session.Player.RPSSelection, opponent.RPSSelection));
+        }
+
+        private static void HandleClaimReward(GameSession session, PacketReader packet)
+        {
+            int rewardTier = packet.ReadInt();
+
+            RPS rpsEvent = DatabaseManager.Events.FindRockPaperScissorsEvent();
+            if (rpsEvent is null)
+            {
+                return;
+            }
+
+            GameEventUserValue rewardsAccumulatedValue = GameEventHelper.GetUserValue(session.Player, rpsEvent.Id, TimeInfo.Tomorrow(), GameEventUserValueType.RPSRewardsClaimed);
+            List<string> rewardsClaimedStrings = rewardsAccumulatedValue.EventValue.Split(",").ToList();
+            foreach (string rewardString in rewardsClaimedStrings)
+            {
+                // User has not claimed any rewards
+                if (rewardString == "")
+                {
+                    break;
+                }
+
+                if (int.TryParse(rewardString, out int rewardInt) && rewardInt == rewardTier)
+                {
+                    return;
+                }
+            }
+
+            RPSTier tier = rpsEvent.Tiers[rewardTier];
```

Comment:
If it tries to get a tier which doesn't exist, it'll crash. Maybe try to use `rpsEvent.Tiers.TryGetValue`

Reply:
or `FirstOrDefault`